kernel/kernel-api/src/main/java/io/delta/kernel/internal/icebergcompat/IcebergCompatV2MetadataValidatorAndUpdater.java

```diff
@@ -131,9 +132,15 @@ public static void validateDataFileStatus(DataFileStatus dataFileStatus) {
               .newMetadata
               .getPartitionColNames()
               .forEach(
-                  partitonCol -> {
+                  partitionCol -> {
```

Comment:
how did we get into a situation where the partition column is not present in the schema?

Is it because we do the schema validations (e.g. valid partition column) later than the icebergCompatV2 validations?